File: nagasena/src/org/openexi/proc/io/BinaryDataSink.java
package org.openexi.proc.io;

import java.io.IOException;

/**
 * BinaryDataSink represents a sink that accepts successive chunks of binary data.
 */
public interface BinaryDataSink {
  
  /** @y.exclude */
  public void startBinaryData(long totalSize, Scribble scribble, Scriber scriber) throws ScriberRuntimeException, IOException;
  
  /** @y.exclude */
  public void binaryData(byte[] byteArray, int startIndex, int length, Scribble scribble, Scriber scriber) throws IOException;
  
  /** @y.exclude */
  public void endBinaryData(Scribble scribble, int localName, int uri, Scriber scriber) throws ScriberRuntimeException, IOException;
  
}

File: nagasenacs/src/Nagasena/Proc/IO/BinaryDataSink.cs
namespace Nagasena.Proc.IO {

  /// <summary>
  /// BinaryDataSink represents a sink that accepts successive chunks of binary data.
  /// </summary>
  public interface BinaryDataSink {

    /// <summary>@y.exclude</summary>
    /// <exception cref="ScriberRuntimeException"></exception>
    void startBinaryData(long totalSize, Scribble scribble, Scriber scriber);

    /// <summary>@y.exclude</summary>
    void binaryData(byte[] byteArray, int startIndex, int length, Scribble scribble, Scriber scriber);

    /// <summary>@y.exclude</summary>
    /// <exception cref="ScriberRuntimeException"></exception>
    void endBinaryData(Scribble scribble, int localName, int uri, Scriber scriber);
  }

}